test/TestApps/PushNotificationsTestApp/main.cpp

```diff
@@ -0,0 +1,349 @@
+    }
+    catch (...)
+    {
+        auto channelRequestException = hresult_error(to_hresult());
+        hr = channelRequestException.code();
+    }
+
+    return (hr == E_INVALIDARG);
+}
+
+HRESULT ChannelRequestHelper(IAsyncOperationWithProgress<PushNotificationCreateChannelResult, PushNotificationCreateChannelStatus> const& channelOperation)
+{
+    if (channelOperation.wait_for(std::chrono::seconds(300)) != AsyncStatus::Completed)
+    {
+        channelOperation.Cancel();
+        return ERROR_TIMEOUT; // timed out or failed
+    }
+
+    auto result = channelOperation.GetResults();
+    auto status = result.Status();
+    if (status != PushNotificationChannelStatus::CompletedSuccess)
+    {
+        return result.ExtendedError(); // did not produce a channel
+    }
+
+    result.Channel().Close();
+    return S_OK;
+}
+
+bool ChannelRequestUsingRemoteId()
+{
+    auto channelOperation = PushNotificationManager::CreateChannelAsync(remoteId1);
+    auto channelOperationResult = ChannelRequestHelper(channelOperation);
+
+    return channelOperationResult == S_OK;
+}
+
+bool MultipleChannelClose()
+{
+    auto channelOperation = PushNotificationManager::CreateChannelAsync(remoteId1);
+    if (channelOperation.wait_for(std::chrono::seconds(300)) != AsyncStatus::Completed)
+    {
+        channelOperation.Cancel();
+        return false; // timed out or failed
+    }
+
+    auto result = channelOperation.GetResults();
+    auto status = result.Status();
+    if (status != PushNotificationChannelStatus::CompletedSuccess)
+    {
+        return false; // did not produce a channel
+    }
+
+    result.Channel().Close();
+    try
+    {
+        result.Channel().Close();
+    }
+    catch (...)
+    {
+        auto channelRequestException = hresult_error(to_hresult());
+        return channelRequestException.code() == WPN_E_CHANNEL_CLOSED;
+    }
+    return false;
+}
+
+bool MultipleChannelRequestUsingSameRemoteId()
+{
+
+    auto channelOperation1 = PushNotificationManager::CreateChannelAsync(remoteId1);
+    auto channelOperation2 = PushNotificationManager::CreateChannelAsync(remoteId1);
+    auto channelOperationResult2 = ChannelRequestHelper(channelOperation2);
+    auto channelOperationResult1 = ChannelRequestHelper(channelOperation1);
+
+    return channelOperationResult2 == S_OK;
+}
+
+bool MultipleChannelRequestUsingMultipleRemoteId()
+{
+    auto channelOperation1 = PushNotificationManager::CreateChannelAsync(remoteId1);
+    auto channelOperation2 = PushNotificationManager::CreateChannelAsync(remoteId2);
+    auto channelOperationResult2 = ChannelRequestHelper(channelOperation2);
+    auto channelOperationResult1 = ChannelRequestHelper(channelOperation1);
+
+    return channelOperationResult2 == S_OK;
+}
+
+bool ActivatorTest()
+{
+    PushNotificationManager::UnregisterActivator(appToken, PushNotificationRegistrationOption::PushTrigger | PushNotificationRegistrationOption::ComActivator);
```

Comment:
Should we also set `appToken` to `nullptr` so we don't accidentally double-unregister if somebody else later looks at the `appToken`?

Reply:
I added std::exchange(appToken, nullptr) to this test to avoid this.